vars/managerPipeline.groovy

```diff
@@ -178,7 +178,87 @@ def call(Map pipelineParams) {
                             wrap([$class: 'BuildUser']) {
                                 dir('scylla-cluster-tests') {
                                     timeout(time: 5, unit: 'MINUTES') {
-                                        createArgusTestRun(params)
```

Comment:
Why can't we consolidate all those into one function ? And not create duplicates of the env handling logic it should be almost identical across the pipeline line.

especially those parts which are tested for existence first.

I think we should get unified, and not create more unique copies of that.

Introducing a new environment variable would be more painful then it's today.

Reply:
the problem is that not identical.  it contains additional logic and different params
for example  [artifactsPipeline.groovy](https://github.com/scylladb/scylla-cluster-tests/pull/6451/files#diff-10408a52873447dcb93fa54ca89448ec42fd97d21136edea55eb798d0c4fa9a8)

dynamically cange instance type in for statement fro not param variable 
```

                                                    if [[ ! -z "${instance_type}" ]]; then
                                                        case "${params.backend}" in
                                                            "aws")
                                                                export SCT_INSTANCE_TYPE_DB="${instance_type}"
                                                                ;;
                                                            "gce")
                                                                export SCT_GCE_INSTANCE_TYPE_DB="${instance_type}"
                                                                ;;
                                                            "azure")
                                                                export SCT_AZURE_INSTANCE_TYPE_DB="${instance_type}"
                                                                ;;
                                                        esac
                                                    fi
```
Tomorrow will continue working on it a bit for now I see the solution  to create 1 function per pipeline 


```

```

Reply:
@fruch, I tried to move it into function. Unsuccessfully

An export variable has gone since we exit SH block.
and moving it to a lower-level bash script will be very problematic since we are using  Groovy objects.